### ISSUE TITLE
Something is wrong with munch and pickle

### ISSUE THREAD

**Issue by harlowja** [Infinidat/munch#30]
An easy test case that shows something is wrong here.

```
>>> m = {'a': {'b': 'c'}}
>>> m = munch.munchify(m)
>>> m
Munch({'a': Munch({'b': 'c'})})
>>> e = munch.munchify(m)
>>> pickle.loads(pickle.dumps(e))
Munch({'a': {'b': 'c'}})
```

I did not expect to lose information when using pickle; but it appears it does.

**Comment by vmalloc** [comment]
@sayag11 I'm already on it. A fix will be available soon

**Comment by vmalloc** [comment]
Fixed and pushed v2.3.1. Should be available on Pypi soon.

**Comment by vmalloc** [comment]
@harlowja thanks for reporting!
